lib/core/datatable/components/datatable/datatable.component.html

```diff
@@ -3,7 +3,7 @@
     *ngIf="data" class="adf-full-width"
     [class.adf-datatable-card]="display === 'gallery'"
     [class.adf-datatable-list]="display === 'list'"
-    [class.adf-sticky-header]="stickyHeader"
+    [class.adf-sticky-header]="stickyHeader && isHeaderVisible()"
```

Comment:
maybe makes more sense adding the "stickyHeader" check inside the "isHeaderVisible()" ? to have only one condition check?